crates/ruff/src/test.rs

```diff
@@ -21,29 +21,38 @@ use crate::registry::AsRule;
 use crate::rules::pycodestyle::rules::syntax_error;
 use crate::settings::{flags, Settings};
 
-pub(crate) fn test_resource_path(path: impl AsRef<Path>) -> std::path::PathBuf {
+pub const DEFAULT_MAX_ITERATIONS: usize = 20;
+
+pub fn test_resource_path(path: impl AsRef<Path>) -> std::path::PathBuf {
     Path::new("./resources/test/").join(path)
 }
 
 /// Run [`check_path`] on a file in the `resources/test/fixtures` directory.
-pub(crate) fn test_path(path: impl AsRef<Path>, settings: &Settings) -> Result<Vec<Message>> {
+pub fn test_path(
+    path: impl AsRef<Path>,
+    settings: &Settings,
+    max_iterations: usize,
+) -> Result<Vec<Message>> {
     let path = test_resource_path("fixtures").join(path);
     let contents = std::fs::read_to_string(&path)?;
-    Ok(test_contents(&contents, &path, settings))
+    Ok(test_contents(&contents, &path, settings, max_iterations))
 }
 
 /// Run [`check_path`] on a snippet of Python code.
-pub(crate) fn test_snippet(contents: &str, settings: &Settings) -> Vec<Message> {
+pub fn test_snippet(contents: &str, settings: &Settings, max_iterations: usize) -> Vec<Message> {
     let path = Path::new("<filename>");
     let contents = dedent(contents);
-    test_contents(&contents, path, settings)
+    test_contents(&contents, path, settings, max_iterations)
 }
 
 /// A convenient wrapper around [`check_path`], that additionally
 /// asserts that autofixes converge after a fixed number of iterations.
-fn test_contents(contents: &str, path: &Path, settings: &Settings) -> Vec<Message> {
-    static MAX_ITERATIONS: usize = 20;
-
+fn test_contents(
+    contents: &str,
+    path: &Path,
+    settings: &Settings,
+    max_iterations: usize,
```

Comment:
Same as for `test_path`. We should extract the logic into a `TestContentsRunner` and use it inside of `test_contents`. This way, it becomes possible for you to use the advanced `runner` API without having to change all call sites (and reduces the number of arguments).